sms_plusserver.py

```diff
@@ -299,17 +322,26 @@ def put_sms(
             data['enc'] = encoding
         max_parts = max_parts or self.max_parts
         if max_parts:
-            data['maxparts'] = six.text_type(max_parts)
+            data['maxparts'] = str(max_parts)
 
         if timeout is None:
             timeout = self.timeout
 
         return self._request(
-            url=self.put_url, data=data, auth=auth, timeout=timeout,
-            resource_name='Put SMS', fail_silently=fail_silently
+            url=self.put_url,
+            data=data,
+            auth=auth,
+            timeout=timeout,
+            resource_name='Put SMS',
+            fail_silently=fail_silently,
         )
 
-    def check_sms_state(self, handle_id, timeout=None, fail_silently=False):
+    def check_sms_state(
+        self,
+        handle_id: str,
+        timeout: Optional[float] = None,
```

Comment:
I'd keep the `Optional` parameters at the end, before the required ones. (or the `fail_silently` is Optional and missing the typehint?)

Here and in the methods below.

Reply:
> ... the `fail_silently` is Optional

Yes, it is but `Optional` is more like `Union` and it is used only to combine `None` and other type.